src/main/java/uk/gov/hmcts/reform/sandl/snlrules/messages/commands/UpdateFactCommand.java

```diff
@@ -14,6 +15,16 @@
 
     @Override
     public void execute(String data) {
-        throw new NotImplementedException("mot implmented yet");
+        KieSession session = droolsService.getRulesSession();
+        Object fact = deserializeMessage(data, this.getFactType());
```

Comment:
I'm tempted to say that we should use `Fact fact = deserializeMessage(data, Fact.class);` here, but it may break later update so I'll leave it up to you.

Reply:
Yip but then Drools uses that as a Fact type not the original type.